lib/ansible/plugins/callback/__init__.py

```diff
@@ -555,11 +555,32 @@ def v2_runner_on_skipped(self, result: TaskResult) -> None:
             host = result._host.get_name()
             self.runner_on_skipped(host, self._get_item_label(getattr(result._result, 'results', {})))
 
-    def v2_runner_on_unreachable(self, result):
+    def v2_runner_on_unreachable(self, result: TaskResult) -> None:
+        """Get details about a task that Ansible could not perform because it could not reach a
+        targeted host, then process the details as required by the callback (output, profiling,
+        logging, notifications, etc.)
+
+        Customization note: For more information about the attributes and methods of the
+        TaskResult class, see lib/ansible/executor/task_result.py.
+
+        :param TaskResult result: An object that contains details about the task
+
+        :return: None
+        """
         host = result._host.get_name()
         self.runner_on_unreachable(host, result._result)
 
-    def v2_runner_on_async_poll(self, result):
+    def v2_runner_on_async_poll(self, result: TaskResult) -> None:
+        """Get details about a task that is running in asynchronous mode and process them as
+        required by the callback (output, profiling, logging, notifications, etc.)
+
+        Customization note: For more information about the attributes and methods of the
+        TaskResult class, see lib/ansible/executor/task_result.py.
+
+        :param TaskResult result: An object that contains details about the task
+
+        :return: None
+        """
         host = result._host.get_name()
         jid = result._result.get('ansible_job_id')
         # FIXME, get real clock
```

Comment:
> _NOTE: Would like more information about the `# FIXME, get real clock` comment in the `v2_runner_on_async_poll` method for future development._

I did some Git paleontology and discovered that this comment was added by @bcoca in https://github.com/ansible/ansible/commit/ba0e5323d6feca04b721ae164e69b68bc1e97b92#diff-4195aff77c2aff536d9383ff218ef4eb114f7a03918c62d607d19a975a53f06aR129 so he's the best person to ask about it…

Reply:
Thanks!

Reply:
iirc this was prepping for when the clock is passed in from the event producer.